kge/util/dump.py

```diff
@@ -113,90 +113,174 @@ def _add_dump_trace_parser(subparsers_dump):
+            " --test is specified)."
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--test",
+        action="store_const",
+        const=True,
+        default=False,
+        help=(
+            "Include entries from evaluation on the test data split (enabled when "
+            "  none of --train, --valid, or --test is specified)."
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--search",
+        action="store_const",
+        const=True,
+        default=False,
+        help="Dump the tracefile of a search job. The options --train, --valid, and"
+             " --test are not applicable.",
+    )
+    parser_dump_trace.add_argument(
+        "--keysfile",
+        default=False,
+        help=(
+            "A path to a file which contains lines in the format"
+            " 'new_key_name'='key_name'. For every line in the keys file, the command"
+            " will look for the value of 'key_name' in the trace entries and config and"
+            " add a respective column in the csv file with name 'new_key_name'."
+            " Additionally, for 'key_name' the special keys '$folder', '$machine'"
+            " '$checkpoint' and '$base_model' can be used. "
+        ),
+    )
+    parser_dump_trace.add_argument(
+        "--keys",
+        "-k",
+        nargs="*",
+        type=str,
+        help=(
+            "A list of 'key' entries (separated by space). Each 'key' has form"
+            " 'new_key_name=key_name' or 'key_name'. This will add columns as in the"
+            " --keysfile option. When only 'key_name' is provided, it is also used as"
+            " column name."
+        ),
+    )
     parser_dump_trace.add_argument(
         "--checkpoint",
         default=False,
         action="store_const",
         const=True,
         help=(
-            "If source is a path to a job folder and --checkpoint is set the best "
-            "(if present) or last checkpoint will be used to determine the job_id"
+            "If source is a path to a job folder and --checkpoint is set, the best"
+            " (if present) or last checkpoint will be used to determine the job_id from"
+            " where the tracefile will be processed backwards."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--job_id",
         default=False,
         help=(
-            "Specifies the training job id in the trace "
-            "from where to start processing backward"
+            "Specifies the training job id in the trace"
+            " from where to start processing the tracefile backward."
         ),
     )
-
     parser_dump_trace.add_argument(
         "--max_epoch",
         default=False,
         help=(
-            "Specifies the epoch in the trace"
-            "from where to start processing backwards"
+            "Specifies the max epoch number in the tracefile"
+            " from where to start processing backwards. Can not be used with --truncate"
```

Comment:
Cannot. What's the point of this option?

Reply:
I think it makes sense. It allows the user to just dump, say, entries of epochs 1-100 independent of a checkpoint or anything else. Apart from that, you asked me to add this option.

Reply:
Why not call this option --truncate_epoch

Reply:
I find --max_epoch (which was your suggestion) more intuitive than --truncate_epoch. But I can change it.

Reply:
How about using only --truncate with an optional numeric argument. If given, then that epoch is used. If not given, then the checkpoint epoch is used (if a checkpoint is specified; otherwise error).

Reply:
But --truncate can only be used together with a checkpoint. --max_epoch is independent of checkpoints.

Reply:
I suggest to change that.

--truncate                   -> use epoch of checkpoint (required)
--truncate 50              -> truncate at epoch 50 (no matter whether or not a checkpoint is specified)

--max-epochs is then not needed anymore.

Reply:
If you want this I can do it (As long as the parser allows for arguments of this kind?).

Reply:
Yes, please go ahead. It's the cleanest and most intuitive way to do this.